### PR TITLE
Application load balancer

### PR DESCRIPTION
**Pull request by msailes**
*Issue #, if available:*
https://github.com/aws/aws-lambda-java-libs/issues/82
https://github.com/aws/aws-lambda-java-libs/issues/90

*Description of changes:*
Added support for the ALB request and response events.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerRequestEvent.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        private Elb elb;
+
+    }
+
+    private RequestContext requestContext;
+    private String httpMethod;
+    private String path;
+    private Map<String, String> queryStringParameters;
+    private Map<String, String> headers;
```
Comment (carlzogh) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerRequestEvent.java:41]: Might as well also add Multi-Value headers https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#multi-value-headers-request 🙂 
```
private Map<String, List<String>> multiValueHeaders;
```

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerResponseEvent.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+@NoArgsConstructor
+@Data
+public class ApplicationLoadBalancerResponseEvent implements Serializable, Cloneable {
+
+    private int statusCode;
+    private String statusDescription;
+    private boolean isBase64Encoded;
+    private Map<String, String> headers;
```
Comment (carlzogh) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerResponseEvent.java:24]: Also might as well also add Multi-Value headers https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#multi-value-headers-response 🙂 
```
private Map<String, List<String>> multiValueHeaders;
```

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerRequestEvent.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * @author msailes <msailes@amazon.co.uk>
+ */
+
+@NoArgsConstructor
+@Data
+public class ApplicationLoadBalancerRequestEvent implements Serializable, Cloneable  {
+
+    @NoArgsConstructor
+    @Data
+    public class Elb implements Serializable, Cloneable {
```
Comment (carlzogh) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerRequestEvent.java:23]: Inner class could be static

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerRequestEvent.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        private String targetGroupArn;
+
+    }
+
+    @NoArgsConstructor
+    @Data
+    public class RequestContext implements Serializable, Cloneable {
```
Comment (carlzogh) [inline aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerRequestEvent.java:31]: Inner class could be static

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerRequestEvent.java (hunk 1 above)
Comment (msailes) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerRequestEvent.java:42]: done

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerResponseEvent.java (hunk 2 above)
Comment (msailes) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerResponseEvent.java:25]: done

File: aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerRequestEvent.java (hunk 3 above)
Comment (msailes) [reply aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/ApplicationLoadBalancerRequestEvent.java]: done

**Pull request merged**